### ISSUE TITLE
TASK-045: Implement LSTM training method

### ISSUE THREAD

**Issue by tomike** [tomike/forecast_aprovizionare#45]
**Epic:** ML Model Development | **Sprint:** 4 (Week 4) | **Priority:** P0

### Description
Implement LSTM model training.

### Tasks
- [ ] Build LSTM layers
- [ ] Compile model (optimizer, loss)
- [ ] Train on sequence data
- [ ] Use validation set for early stopping

### Acceptance Criteria
- Model trains successfully
- Validation is performed
- Training converges
